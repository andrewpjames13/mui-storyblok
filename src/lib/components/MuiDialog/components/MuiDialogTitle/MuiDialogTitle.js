import React from 'react';
import PropTypes from 'prop-types';
import DialogTitle from '@material-ui/core/DialogTitle';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import Storyblok from '../../../../utils/Storyblok';

const MuiDialogTitle = ({
  toggleDialog,
  rootClass,
  text,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <DialogTitle
      disableTypography
      className={styles.root}
    >
      {text}
      <IconButton aria-label="close" onClick={toggleDialog}>
        <CloseIcon />
      </IconButton>
    </DialogTitle>
  );
};

export default MuiDialogTitle;

MuiDialogTitle.propTypes = {
  text: PropTypes.string.isRequired,
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /** passed down from parent to close Dialog */
  toggleDialog: PropTypes.func.isRequired,
};

MuiDialogTitle.defaultProps = {
  rootClass: [],
};
