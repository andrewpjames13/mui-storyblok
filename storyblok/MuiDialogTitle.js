module.exports = {
  "name": "MuiDialogTitle",
  "display_name": null,
  "updated_at": "2020-03-31T18:27:06.714Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "max_length": ""
    },
    "text": {
      "type": "text",
      "required": true
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiDialogTitle",
  "component_group_uuid": null
}