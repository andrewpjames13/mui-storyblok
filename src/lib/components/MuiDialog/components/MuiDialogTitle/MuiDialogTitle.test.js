import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiDialogTitle from './MuiDialogTitle';

function setup() {
  const props = {
    text: 'text',
    toggleDialog: jest.fn(),
  };
  const comp = shallow(<MuiDialogTitle {...props} />);

  return { comp, props };
}

describe('<MuiDialogTitle />', () => {
  it('renders MuiDialogTitle', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });


  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<MuiDialogTitle {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
