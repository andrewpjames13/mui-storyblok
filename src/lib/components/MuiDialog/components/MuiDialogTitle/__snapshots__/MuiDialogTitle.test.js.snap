// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiDialogTitle /> snapshot 1`] = `
<div
  className="MuiDialogTitle-root makeStyles-root-3"
>
  text
  <button
    aria-label="close"
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
  </button>
</div>
`;
